update_agol_vehicles_pallet.py

```diff
@@ -0,0 +1,201 @@
+    for layer in sharing_map.listLayers():
+        log.info(f'Removing {layer} from {sharing_map.name}...')
+        sharing_map.removeLayer(layer)
+
+    log.info(f'Adding {fc_to_add} as layer to {sharing_map.name}...')
+    layer = sharing_map.addDataFromPath(fc_to_add)
+    project.save()
+
+    return layer, sharing_map
+
+
+def update_agol_feature_service(sharing_map, layer, feature_service_name,
+                                sddraft_path, sd_path, sd_item):
+    '''
+    Helper method for updating an AGOL hosted feature service from an ArcGIS
+    Pro arcpy.mp.Map and .Layer objects.
+
+    sharing_map:            An arcpy.mp.Map object containing the layer to be
+                            shared.
+    layer:                  The arcpy.mp.Layer object created from the feature
+                            class that holds your new data.
+    feature_service_name:   The name of the existing Hosted Feature Service.
+                            Must match exactly, otherwise the publish step will
+                            fail.
+    sddraft_path, sd_path:  Strings of the paths to save the service definition
+                            draft and final files.
+    sd_item:                The URL of the service definition item on AGOL
+                            originally used to publish the hosted feature
+                            service.
+    '''
+
+    sharing_draft = sharing_map.getWebLayerSharingDraft(
+        'HOSTING_SERVER', 'FEATURE', feature_service_name, [layer])
+    sharing_draft.exportToSDDraft(sddraft_path)
+    arcpy.server.StageService(sddraft_path, sd_path)
+    sd_item.update(data=sd_path)
+    sd_item.publish(overwrite=True)
+
+class AGOLVehiclesPallet(Pallet):
+    def requires_processing(self):
+        #: No crates, run process every time
+        return True
+
+    def process(self):
+
+        #: Set up paths and directories
+        feature_service_name = secrets.FEATURE_SERVICE_NAME
+
+        temp_csv_dir = os.path.join(arcpy.env.scratchFolder, 'fleet')
+        temp_fc_path = os.path.join(arcpy.env.scratchGDB, feature_service_name)
+        sddraft_path = os.path.join(arcpy.env.scratchFolder,
+                                    f'{feature_service_name}.sddraft')
+        sd_path = sddraft_path[:-5]
+
+        paths = [temp_csv_dir, temp_fc_path, sddraft_path, sd_path]
+        for item in paths:
+            if arcpy.Exists(item):
+                self.log.info(f'Deleting {item} prior to use...')
+                arcpy.Delete_management(item)
+        os.mkdir(temp_csv_dir)
+
+        if not secrets.KNOWNHOSTS or not os.path.isfile(secrets.KNOWNHOSTS):
+            raise FileNotFoundError(
+                f'known_hosts file {secrets.KNOWNHOSTS} not found. Please '
+                'create with ssh-keyscan.')
+
+        #: Download all the files in the upload folder on sftp to temp_csv_dir
+        self.log.info(
+            f'Downloading all files from {secrets.SFTP_HOST}/upload...')
```

Comment:
**thought**: in this case logging a secret is probably ok, but keep this in mind in the future, especially if we are log shipping to somewhere else.